### DISCUSSION TITLE
trying to set fixed IP on vm, but failed.

### DISCUSSION THREAD

**Discussion by ronnin** [lima-vm/lima#677] [Q&A] [answered]
Here is my lima provision configuration:
`
  - location: "https://cloud-images.ubuntu.com/impish/current/impish-server-cloudimg-amd64.img"
     arch: "x86_64"
  - location: "https://cloud-images.ubuntu.com/impish/current/impish-server-cloudimg-arm64.img"
     arch: "aarch64"
  mounts:
    - location: "~"
    - location: "/tmp/lima"
       writable: true
  provision:
    - mode: system
       script: |
           #!/bin/bash
           set -eux -o pipefail
           cat <<EOF | sudo tee /etc/cloud/cloud.cfg.d/01_static_network.cfg
           network:
              version: 2
              ethernets:
              eth0:
                 dhcp4: false
                 addresses:
                    - 192.168.5.11/24
                 gateway4: 192.168.5.3
                 nameservers:
                 addresses:
                    - 192.168.5.3
                 set-name: eth0
           EOF
`
I have no idea about getting the mac address of eth0 in the 'to-be-provision' vm.
So, the configuration of netplan in the cloud-init script shown above didn't contains the math/macaddress part.

With the confiuration, new vm could be privisioned, but its IP was still 192.168.5.15, even if **/etc/netplan/50-cloud-init.yaml** was checked out as the following:
`
   network:
       ethernets:
          eth0:
             addresses:
                - 192.168.5.11/24
             dhcp4: false
             gateway4: 192.168.5.3
             nameservers:
                addresses:
                   - 192.168.5.3
              set-name: eth0
      version: 2
`

**Comment by jandubois** [comment, marked as answer]
It is not possible to change the internal IP address. Is there a reason why you want to do this? The address is not routable from anywhere and only valid inside the guest, so why does it matter which address it has?

**Reply by ronnin** [reply]
Thanks for so rapid answer.

I want to install a 3-node k8s cluster, such as kubesphere.

**Reply by jandubois** [reply]
> I want to install a 3-node k8s cluster, such as kubesphere.

To have routable IP addresses for your instances, you need to use [vde_vmnet](https://github.com/lima-vm/lima/blob/master/docs/network.md#managed-vmnet-networks-via-vde_vmnet).
